### PR TITLE
V4 constants change overhaul

### PR DESCRIPTION
**Pull request by SEAJang**
# Pull Request Key Information

## Reason for Change

In the codebase, locating constants was challenging, and there were instances where hard-coded values led to potential errors. To mitigate this issue and enhance code maintainability, introduced the usage of constants. Moreover, consolidated all constants into a single module for easy access.

## Changes Made

- Utilized constants where applicable to minimize errors stemming from hard-coded values.
- Migrated all constants to a dedicated module for streamlined access.

## Objective

The primary objective of these changes is to enhance code reliability by reducing potential errors and improving code readability and navigability.

## Testing Undertaken
All existing unit tests passed and performed manual simulations
[results.txt](https://github.com/LDAR-Sim/LDAR_Sim/files/14924150/results.txt)

### CODE REVIEW

File: LDAR_Sim/src/constants/file_name_constants.py
Comment (ThomasGalesloot) [inline LDAR_Sim/src/constants/file_name_constants.py:1]: Recommend either combining these with the output file constants and renaming to generic file constants or moving the program outputs filenames from this file to those constants. Also recommend organizing the program outputs filenames into some class together

File: LDAR_Sim/src/constants/general_const.py
Comment (ThomasGalesloot) [inline LDAR_Sim/src/constants/general_const.py:1]: Emissions and placeholder constants don't feel like they belong here

File: LDAR_Sim/testing/unit_testing/test_virtual_world/test_emissions/emissions_testing_fixtures.py
Comment (ThomasGalesloot) [inline LDAR_Sim/testing/unit_testing/test_virtual_world/test_emissions/emissions_testing_fixtures.py:1]: Use constants here as well

File: LDAR_Sim/testing/unit_testing/test_virtual_world/test_emissions/test_update.py
Comment (ThomasGalesloot) [inline LDAR_Sim/testing/unit_testing/test_virtual_world/test_emissions/test_update.py:1]: Use constants here as well

File: LDAR_Sim/testing/unit_testing/test_virtual_world/test_fugitive_emission/fugitive_emission_testing_fixtures.py
Comment (ThomasGalesloot) [inline LDAR_Sim/testing/unit_testing/test_virtual_world/test_fugitive_emission/fugitive_emission_testing_fixtures.py:1]: use constants here as well

File: LDAR_Sim/testing/unit_testing/test_virtual_world/test_fugitive_emission/test_activate.py
Comment (ThomasGalesloot) [inline LDAR_Sim/testing/unit_testing/test_virtual_world/test_fugitive_emission/test_activate.py:1]: Use constants here as well

File: LDAR_Sim/testing/unit_testing/test_virtual_world/test_nonfugitive_emissions/nonfugitive_emission_testing_fixtures.py
Comment (ThomasGalesloot) [inline LDAR_Sim/testing/unit_testing/test_virtual_world/test_nonfugitive_emissions/nonfugitive_emission_testing_fixtures.py:1]: Use constants here as well

**Review by ThomasGalesloot — CHANGES_REQUESTED** [review]
Great Work - See my comments for a few minor changes. Also please change all constant classes to be data classes to reduce overhead when we make changes to instantiate them. Also please upload console output from running all unit tests. You can pipe console output from pytest to a file using pytest .\LDAR_Sim\testing > .\LDAR_Sim\testing\end_to_end_testing\test_results\unit_test_results.txt, for example.

File: LDAR_Sim/src/constants/general_const.py
Comment (SEAJang) [reply LDAR_Sim/src/constants/general_const.py:1]: They didn't really fit very well into the other constants, and I don't think a separate file is really needed for these, which was why they were added here.

**Comment by SEAJang** [comment]
[results.txt](https://github.com/LDAR-Sim/LDAR_Sim/files/14924940/results.txt)
Unit test results after initial PR review updates/changes.

**Comment by SEAJang** [comment]
[results.txt](https://github.com/LDAR-Sim/LDAR_Sim/files/14924979/results.txt)
Unit test results.

File: LDAR_Sim/src/constants/file_name_constants.py
Comment (ThomasGalesloot) [inline LDAR_Sim/src/constants/file_name_constants.py:1]: Add dataclass annotations

**Review by ThomasGalesloot — APPROVED** [review]
Great work! - 1 last thing not critical just a nit pick - Could add the dataclass annotation in front of all these classes

File: LDAR_Sim/src/constants/file_name_constants.py
Comment (SEAJang) [reply LDAR_Sim/src/constants/file_name_constants.py:1]: Added to relevant classes...

**Pull request merged**